### PR TITLE
docs: adds channel background customization guide

### PR DESCRIPTION
**Pull request by vanGalilea**
## 🎯 Goal

Adding more customization possibilities to our documentation. 
This PR touches Channel background related customization

## 🛠 Implementation details

2 custom. steps one with a static background and 2nd with a dedicated channel preference that are persisted to a key-value store.

## 🎨 UI Changes

N/A

## 🧪 Testing

pull PR and run `npx stream-chat-docusaurus -i -s`

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [ ] Screenshots added for visual changes
- [x] Documentation is updated

### CODE REVIEW

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+```tsx
+import { Channel, MessageInput, MessageList, ThemeProvider } from 'stream-chat-react-native';
+import { ImageBackground } from 'react-native';
+
+const IMAGE_URI =
+  'https://images.unsplash.com/photo-1549125764-91425ca48850?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8NjF8fHxlbnwwfHx8fA%3D%3D&auto=format&fit=crop&w=800&q=60';
+
+const ChannelScreen = ({ channel }) => (
+  <ThemeProvider style={theme}>
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:32]: where does the `theme` come from? we could add a small example theme for a bit more clarity

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 2]
```diff
@@ -0,0 +1,318 @@
+---
+id: channel-background-customization
+sidebar_position: 13
+title: Channel Background Customization
+---
+
+You can implement this customization in your existing app or create a new one by running the following
+
+```shell
+npx react-native init MyApp --template stream-chat-react-native-template
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:10]: This template is not actively maintained by us. So I am a bit sceptical about recommending this. In my view, we can avoid this section. People would normally follow installing steps first fo through the tutorial and come to this guide when they want some customisation.

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+To persist the channel preferences (at the moment, only the URI of the selected background), we will need to store some data.
+We will do it by using `react-native-mmkv`, a key-value storage [framework](https://github.com/mrousavy/react-native-mmkv#mmkv).
+
+Follow the [installation steps](https://github.com/mrousavy/react-native-mmkv#installation), and let's get started, shall we?
+
+We will start by creating our `ChannelBackgroundView` component.
+This component will be in charge of rendering the custom background by retrieving it from our key-value store.
+We save an object to a key-value store to be scalable and future-proof.
+You'd might want to add other preferences later, such as dimming value, background color, etc.
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:93]:
```suggestion
You'd want to add other preferences later, such as dimming value, background colour, etc.
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+const DEFAULT_BACKGROUND_URI = 'https://i.redd.it/3jfjc53fsyb61.jpg';
+
+const ChannelBackgroundView = ({
+  channelId,
+  ...props
+}: {
+  channelId: string;
+} & ViewProps) => {
+  const [channelPreferences] = useMMKVObject<ChannelPreferences>(channelId);
+  let uri: string | undefined = channelPreferences?.imageUri || DEFAULT_BACKGROUND_URI;
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:112]:
```suggestion
  const uri: string | undefined = channelPreferences?.imageUri || DEFAULT_BACKGROUND_URI;
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+
+  return <ImageBackground {...props} source={{ uri }} />;
+};
+```
+
+We will then use it in our previously built `ChannelScreen`.
+Replace the static `ImageBackground` with `ChannelBackgroundView` and pass the `channelId`
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:119]:
```suggestion
Replace the static `ImageBackground` with `ChannelBackgroundView` and pass the `channelId`.
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+```
+
+:::note
+Be aware of the fact that channel preferences was implemented with MMKV, a key-value storage framework.
+There are alternative approaches to achieve the same goal, for example by saving the channel preferences as [custom data](https://getstream.io/chat/docs/javascript/channel_update/?language=javascript) on Stream's channel object.
+:::
+
+### Add a configuration button
+
+We will add now a button that will take the user from the Channel screen to our new WallpaperOverview screen
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:243]:
```suggestion
We will add now a button that will take the user from the Channel screen to our new WallpaperOverview screen.
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+    },
+  },
+};
+```
+
+---
+
+### Optional: Connect all screens by navigation
+
+If applicable to your use case, add our screens to a Navigation Stack by doing the following.
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:296]:
```suggestion
If applicable to your use case, add our screens to a Navigation Stack by doing the following:
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx (hunk 2 above)
Comment (vishalnarkhede) [reply docusaurus/docs/reactnative/guides/channel_background_customization.mdx:10]: Agree on this one!!

Comment (vishalnarkhede) [reply docusaurus/docs/reactnative/guides/channel_background_customization.mdx:10]: Someone reading this section already has an app running on his machine ... either his own app or from tutorial

Comment (madsroskar) [reply docusaurus/docs/reactnative/guides/channel_background_customization.mdx:10]: Welllll, it's maintained by me 😅

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx (hunk 3 above)
Comment (madsroskar) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:93]:
```suggestion
To persist the channel preferences (at the moment, only the URI of the selected background), we will need to store some data.
We will do it by using `react-native-mmkv`, a key-value storage [framework](https://github.com/mrousavy/react-native-mmkv#mmkv).

Follow the [installation steps](https://github.com/mrousavy/react-native-mmkv#installation), and let's get started, shall we?

We will start by creating our `ChannelBackgroundView` component.
This component will be in charge of rendering the custom background by retrieving it from our key-value store.
We save an object to a key-value store to be scalable and future-proof.
You might want to add other preferences later, such as dimming value, background color, etc.
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx (hunk 4 above)
Comment (madsroskar) [reply docusaurus/docs/reactnative/guides/channel_background_customization.mdx:112]: Isn't the `| undefined` unnecessary here? I'd also avoid explicitly typing things as `undefined` and rather use something more explicit like `null`

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx (hunk 6 above)
Comment (madsroskar) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:243]:
```suggestion
We will now add a button that will take the user from the Channel screen to our new WallpaperOverview screen
```

**Review by madsroskar — CHANGES_REQUESTED** [review]
Only comment I think _must_ be fixed is the unnecessary gendering of the user - the others are more nit-y.

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+
+You can implement this customization in your existing app or create a new one by running the following
+
+```shell
+npx react-native init MyApp --template stream-chat-react-native-template
+```
+
+## Basic Custom Background
+
+You can change the background statically by simply wrapping `MessageList` and `MessageInput` by using the `ImageBackground` component.
```
[the remark this answers is not in the document]
Comment (santhoshvai) [reply docusaurus/docs/reactnative/guides/channel_background_customization.mdx:15]: That's a nice reference @madsroskar, I am completely unaware of it

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx (hunk 1 above)
Comment (vanGalilea) [reply docusaurus/docs/reactnative/guides/channel_background_customization.mdx:32]: from line 47, I've moved to the top, so it'll be more clear

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 9]
```diff
@@ -0,0 +1,312 @@
+---
+id: channel-background-customization
+sidebar_position: 13
+title: Channel Background Customization
+---
+
+## Basic Custom Background
+
+You can change the background statically by wrapping `MessageList` and `MessageInput` by using the `ImageBackground` component.
```
Comment (madsroskar) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:9]:
```suggestion
You can change the background statically by wrapping `MessageList` and `MessageInput` using the `ImageBackground` component.
```

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+  'https://images.unsplash.com/photo-1553114835-6f7674d3c2c0?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MTMyfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
+  'https://images.unsplash.com/photo-1553075712-453f7213c24f?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MTMzfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
+  'https://images.unsplash.com/photo-1551917951-148edcd8ea8d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MTU3fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
+  'https://images.unsplash.com/photo-1553969923-bbf0cac2666b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MjA3fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
+  'https://images.unsplash.com/photo-1553194642-29b272a173b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MTcwfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
+  'https://images.unsplash.com/photo-1553356084-58ef4a67b2a7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MTcxfHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
+  'https://images.unsplash.com/photo-1553526777-5ffa3b3248d8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MTk4fHx8ZW58MHx8fHw%3D&auto=format&fit=crop&w=800&q=60',
```
Comment (madsroskar) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:210]: Should we add some attribution to the images we use? Not required with unsplash, but probably a nice thing to do for the photographers

File: docusaurus/docs/reactnative/guides/channel_background_customization.mdx [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+  image: {
+    flex: 1,
+    width: '100%',
+  },
+});
+```
+
+:::note
+Be aware of the fact that channel preferences was implemented with MMKV, a key-value storage framework.
+There are alternative approaches to achieve the same goal, for example by saving the channel preferences as [custom data](https://getstream.io/chat/docs/javascript/channel_update/?language=javascript) on Stream's channel object.
```
Comment (madsroskar) [inline docusaurus/docs/reactnative/guides/channel_background_customization.mdx:232]:
```suggestion
Be aware of the fact that channel preferences were implemented with MMKV, a key-value storage framework.
There are alternative approaches to achieving the same goal, such as saving the channel preferences as [custom data](https://getstream.io/chat/docs/javascript/channel_update/?language=javascript) on Stream's channel object.
```

**Review by madsroskar — APPROVED** [review]
I've added some nit-y suggestions related to grammar, but not super important. 👍🏻

**Pull request merged**